### PR TITLE
Fix acl/test_stress_acl.py invalid interface name

### PR DESCRIPTION
**Pull request by justin-wong-ce**
### Description of PR
Fix acl/test_stress_acl.py using bad interface name for ACL table creation

Summary:
Fixes # (issue)
In `acl/test_stress_acl.py`, it attempts to retrieve an interface that can be used to create a ACL table. DUTs with and without PortChannels require different methods respectively.

Currently, it checks by filtering with topo. However, some topology flags can have configurations that have or not have PortChannels, making topos no longer a sufficient check - in some topos the test will fail with:
```
Error: Failed to parse ACL table config: exception=Cannot bind ACL to specified port Ethernet136
```

Reproducible by manually running the following on the DUT:
```
config acl add table DATAACL L3 -s ingress -p Ethernet0
^FAILS
config acl add table DATAACL L3 -s ingress -p PortChannel101
^WORKS
```

### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [x] 202405

#### How did you do it?
Fix by checking if a PortChannel exists. If it does - use it. If it does not - fallback on the secondary method to retrieve a normal interface name if its a not a dualtor topo (due to
https://github.com/sonic-net/sonic-mgmt/pull/6960).

#### How did you verify/test it?
Test no longer fails creating an ACL table on a t1 topo with PortChannels.
Tested with Arista HwSkus on t0, t1, t2, and mx topologies.

### CODE REVIEW

**Comment by StormLiangMS** [comment]
hi @bingwang-ms could you help to take a look?

**Review by StormLiangMS — DISMISSED** [review]
LGTM

**Comment by bingwang-ms** [comment]
@justin-wong-ce Can you help me understand why there is no portchannel for test on dualtor platform?

**Comment by justin-wong-ce** [comment]
> @justin-wong-ce Can you help me understand why there is no portchannel for test on dualtor platform?

I decided to handle dualtor differently because of this previous pull request: https://github.com/sonic-net/sonic-mgmt/pull/6960

**Comment by bingwang-ms** [comment]
> > @justin-wong-ce Can you help me understand why there is no portchannel for test on dualtor platform?
> 
> I decided to handle dualtor differently because of this previous pull request: #6960:
> 
> ```
> What is the motivation for this PR?
> In acl stress test, ACL table creation can be failed on dualtor platform because of insufficient resources
> The test case only add/delete a same rule, which is not the real scenario in production
> ```
> 
> I am not too familiar with PortChannels on dualtor but I think sometimes it is not available based on the PR adding dualtor support to the test.

Thanks for the reply. Portchannels should always be available on dualtor platform. 
@xwjiang-ms Can you please help review & comment?

File: tests/acl/test_stress_acl.py [hunk 1]
```diff
@@ -91,12 +91,15 @@ def prepare_test_file(rand_selected_dut):
 @pytest.fixture(scope='module')
 def prepare_test_port(rand_selected_dut, tbinfo):
     mg_facts = rand_selected_dut.get_extended_minigraph_facts(tbinfo)
-    if tbinfo["topo"]["type"] == "mx":
-        dut_port = mg_facts["minigraph_acls"]["DataAcl"][0]
-    else:
-        dut_port = list(mg_facts['minigraph_portchannels'].keys())[0]
+
+    ports = list(mg_facts['minigraph_portchannels'])
```
Comment (xwjiang-ms) [inline tests/acl/test_stress_acl.py:95]: mx topo don't have portchannel

File: tests/acl/test_stress_acl.py [hunk 2]
```diff
@@ -91,12 +91,15 @@ def prepare_test_file(rand_selected_dut):
...
+    if tbinfo["topo"]["type"] != "dualtor" and not ports:
+        ports = mg_facts["minigraph_acls"]["DataAcl"]
+
+    dut_port = ports[0] if ports else None
+
     if not dut_port:
-        pytest.skip('No portchannels found')
+        pytest.skip('No portchannels available in dualtor topo or no interfaces found on DUT')
```
Comment (xwjiang-ms) [inline tests/acl/test_stress_acl.py:102]: I think it's unnecessary to add `in dualtor topo` because it's a common message

**Review by xwjiang-ms — CHANGES_REQUESTED** [review]
LGTM

**Review by xwjiang-ms — APPROVED** [review]

File: tests/acl/test_stress_acl.py (hunk 1 above)
Comment (StormLiangMS) [inline tests/acl/test_stress_acl.py:95]: hi @justin-wong-ce I had a discussion with Xiawei, quote from his response
"""
The purpose is to bind acl to a upstream port, both pc or eth should work, I remember when I write the case, some tb do not have data acl table, so I used pc, but I checked again this afternoon and found t0/t1/dualtor/mx all have data acl table
"""
the port selection logic could be, take pc first, if no, check the dataacl ports. But there should be no difference for dualtor, could we remove dualtor condition check from line 96, it is a little confusing.

**Review by StormLiangMS — APPROVED** [review]
LGTM

**Comment by nhe-NV** [comment]
@r12f  This PR also needed for the 202412 branch, can you help to merge it also to 202412 branch?

**Comment by r12f** [comment]
already included in 202411, removing 202412 tag.

**Pull request merged**